plugins/rest/src/server_command.cpp

```diff
@@ -0,0 +1,312 @@
+  auto rest_options = caf::get_or(inv.options, "rest", caf::settings{});
+  auto data = vast::data{};
+  bool success = convert(rest_options, data);
+  if (!success)
+    return caf::make_message(
+      caf::make_error(ec::invalid_argument, "couldnt parse options"));
+  rest::configuration config;
+  caf::error error = convert(data, config);
+  if (error)
+    return caf::make_message(
+      caf::make_error(ec::invalid_argument, "couldnt convert options"));
+  VAST_INFO("options: {}\ndata: {}", inv.options, data);
+  auto server_config = convert_and_validate(config);
+  if (!server_config)
+    return caf::make_message(caf::make_error(
+      ec::invalid_configuration,
+      fmt::format("invalid server configuration: {}", server_config.error())));
+  // Create necessary actors.
+  auto node_opt = vast::system::spawn_or_connect_to_node(
+    self, inv.options, content(system.config()));
+  if (auto* err = std::get_if<caf::error>(&node_opt))
+    return caf::make_message(std::move(*err));
+  const auto& node
+    = std::holds_alternative<system::node_actor>(node_opt)
+        ? std::get<system::node_actor>(node_opt)
+        : std::get<scope_linked<system::node_actor>>(node_opt).get();
+  VAST_ASSERT(node != nullptr);
+  auto authenticator = get_authenticator(self, node, caf::infinite);
+  if (!authenticator)
+    return caf::make_message(std::move(authenticator.error()));
+  auto dispatcher
+    = self->spawn(request_dispatcher, *server_config, *authenticator);
+  // Set up router.
+  auto router = std::make_unique<router_t>();
+  VAST_ASSERT_CHEAP(dispatcher);
+  // Set up API routes from plugins.
+  for (auto const* rest_plugin : plugins::get<rest_endpoint_plugin>()) {
+    auto prefix = rest_plugin->prefix();
+    if (!prefix.empty() && prefix[0] != '/') {
+      VAST_WARN("ignoring endpoints with invalid prefix {}", prefix);
+      continue;
+    }
+    auto handler = rest_plugin->handler(system, node);
+    for (auto const& endpoint : rest_plugin->api_endpoints()) {
+      if (endpoint.path.empty() || endpoint.path[0] != '/') {
+        VAST_WARN("ignoring route {} due to missing '/'", endpoint.path);
+        continue;
+      }
+      setup_route(self, router, rest_plugin->prefix(), dispatcher,
+                  std::move(endpoint), handler);
+    }
+  }
+  // Set up non-API routes.
+  router->non_matched_request_handler([](auto req) {
+    VAST_VERBOSE("404 not found: {}", req->header().path());
+    return req->create_response(restinio::status_not_found()).done();
+  });
+  router->http_get(
+    "/", [](auto request, auto) -> restinio::request_handling_status_t {
+      return request->create_response()
+        .append_header(restinio::http_field::server, "VAST")
+        .append_header_date_field()
+        .append_header(restinio::http_field::content_type, "text/html; "
+                                                           "charset=utf-8")
+        .set_body(restinio::sendfile("../plugins/rest/www/status.html"))
```

Comment:
This will only work when running VAST from the build tree. We should install the www directory somewhere into DATAROOTDIR, I think.

Reply:
I've removed the html part for now, we can add it back in a later commit.